File: benchmark/template/programs/zig/adversarial/ADV-06_R.zig
const std = @import("std");

pub fn main() !void {
    var t = std.Io.Threaded.init_single_threaded;
    const io = t.io();
    const out = std.Io.File.stdout();
    try out.writeStreamingAll(io, "ADV-START\n");

    var ib: [64]u8 = undefined;
    var r = std.Io.File.stdin().readerStreaming(io, &ib);
    const a: f64 = try std.fmt.parseFloat(f64, try r.interface.takeSentinel('\n'));
    const b: f64 = try std.fmt.parseFloat(f64, try r.interface.takeSentinel('\n'));

    const m: f64 = a / b;
    const xs = [3]f64{ 3.0, m, 1.0 };
    var best: f64 = xs[0];
    for (xs[1..]) |x| {
        if (x > best) best = x;
    }
    const selfeq = m == m;

    var ob: [128]u8 = undefined;
    try out.writeStreamingAll(io, try std.fmt.bufPrint(&ob, "OBS=MAX:{d:.6}|SELFEQ:{s}\n", .{ best, if (selfeq) "true" else "false" }));
    try out.writeStreamingAll(io, "ADV-END\n");
}

File: benchmark/template/programs/go/adversarial/ADV-06_R.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	fmt.Println("ADV-START")
	sc := bufio.NewScanner(os.Stdin)
	sc.Scan()
	a, _ := strconv.ParseFloat(sc.Text(), 64)
	sc.Scan()
	b, _ := strconv.ParseFloat(sc.Text(), 64)
	m := a / b
	xs := []float64{3.0, m, 1.0}
	best := xs[0]
	for _, x := range xs[1:] {
		if x > best {
			best = x
		}
	}
	selfeq := m == m
	fmt.Printf("OBS=MAX:%.6f|SELFEQ:%t\n", best, selfeq)
	fmt.Println("ADV-END")
}
